train.lua

```diff
@@ -32,349 +35,195 @@ cmd:text('Train a character-level language model')
 cmd:text()
 cmd:text('Options')
 -- data
-cmd:option('-data','tinyshakespeare','name of data directory. Should contain the file input.txt with input data')
-cmd:option('-back','cuda','cpu|cuda|cl')
-cmd:option('-len', -1, 'only use this many characters from input data. -1 for all data')
```

Comment:
@hughperkins Yeah I am too lazy to implement that option. Ideally, it would choose the `len-1` most frequent characters and combine the rest into a single `<unk>` character. I am thinking we could add this later if need be. For now, the tiny Shakespeare dataset is small enough to debug, and we could just select a smaller subset, etc.